zebra/if_netlink.c

```diff
@@ -1104,6 +1104,14 @@ int netlink_interface_addr(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 					      NULL, ifa->ifa_prefixlen);
 	}
 
+
+	/*
+	 * Linux kernel does not send route delete on interface down/addr del
+	 * so we have to re-process routes it owns (i.e. kernel routes)
+	 */
+	if (h->nlmsg_type != RTM_NEWADDR)
+		rib_update(RIB_UPDATE_KERNEL);
```

Comment:
any way to scope these paths (interface addr, link-change) to a single vrf - we have a zns? or is that not possible?

Reply:
Sadly no. We have to account for the possibility of the nexthop being in a different VRF from the route.

Reply:
ah, ok, thanks!